pkg/storage/couchdb/utils.go

```diff
@@ -95,41 +96,106 @@ func (db *couchDBQuerier[E]) CreateBasicCounterView() error {
 	return nil
 }
 
-func (db *couchDBQuerier[E]) Count() (int, error) {
-	opts := map[string]interface{}{}
-	rows := db.Query(context.TODO(), "_design/utils", "_view/count", opts)
-	if err := rows.Err(); err != nil {
-		return -1, err
+func (db *couchDBQuerier[E]) EnsureIndexExists(field string) {
+	exists, _, err := db.SelectExists("_design/" + field)
+	if err != nil {
+		fmt.Printf("Error checking for index existence: %s\n", err)
+	}
+	if !exists {
+		db.CreateIndex(field)
+	}
+}
+
+func (db *couchDBQuerier[E]) CreateIndex(fields string) error {
+	querier := newCouchDBQuerier[map[string]interface{}](db.DB)
+
+	indexPayload := map[string]interface{}{
+		"language": "query",
+		"views": map[string]interface{}{
+			fields + "-json-index": map[string]interface{}{
+				"map": map[string]interface{}{
+					"fields": map[string]string{
+						fields: "asc",
+					},
+					"partial_filter_selector": map[string]interface{}{},
+				},
+				"reduce": "_count",
+				"options": map[string]interface{}{
+					"def": map[string]interface{}{
+						"fields": []string{fields},
+					},
+				},
+			},
+		},
 	}
 
-	rows.Next()
-	var result int
-	err := rows.ScanValue(&result)
+	_, err := querier.DB.Put(context.Background(), "_design/"+fields, indexPayload)
 	if err != nil {
-		return -1, err
+		return fmt.Errorf("error creating index: %s", err)
 	}
 
-	return result, nil
+	return nil
+}
+
+func (db *couchDBQuerier[E]) Count(opts *map[string]interface{}) (int, error) {
+
+	if opts == nil {
+		countRows := db.AllDocs(context.Background(), map[string]interface{}{"include_docs": false})
+
+		docCount := 0
+		for countRows.Next() {
+			if !strings.HasPrefix(countRows.ID(), "_design/") {
+				docCount++
+			}
+		}
+		if countRows.Err() != nil {
+			return -1, fmt.Errorf("error while iterating documents: %s", countRows.Err())
+		}
+		defer func() {
+			countRows.Close()
+		}()
+
+		return docCount, nil
+	} else {
+		countRowsBy := db.Find(context.Background(), opts)
```

Comment:
Find operation is not suitable for counting. Find is limited by default to 25 docs. 

We should use views with reduce operations. I thin'k that we should consider that counting is not available in NoSQL storage engines.